From dc95e66b7a1e9ab9c5e9fb892012ada22eb42610 Mon Sep 17 00:00:00 2001
From: Felix Geyer <felix.geyer@credativ.de>
Date: Fri, 15 Sep 2017 13:31:04 +0200
Subject: [PATCH 16/16] Fix XSA-233 patch to use list_for_each_entry.

---
 tools/xenstore/xenstored_domain.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/xenstore/xenstored_domain.c b/tools/xenstore/xenstored_domain.c
index 862395b79..2106d52b8 100644
--- a/tools/xenstore/xenstored_domain.c
+++ b/tools/xenstore/xenstored_domain.c
@@ -185,11 +185,11 @@ static int destroy_domain(void *_domain)
 static void domain_cleanup(void)
 {
 	xc_dominfo_t dominfo;
-	struct domain *domain, *tmp;
+	struct domain *domain;
 	int notify = 0;
 
 again:
-	list_for_each_entry_safe(domain, tmp, &domains, list) {
+	list_for_each_entry(domain, &domains, list) {
 		if (xc_domain_getinfo(*xc_handle, domain->domid, 1,
 				      &dominfo) == 1 &&
 		    dominfo.domid == domain->domid) {
-- 
2.14.1

